Code/Legacy/CrySystem/XML/xml.cpp

```diff
@@ -26,30 +27,7 @@
 
 #include "System.h"
 
-#if AZ_TRAIT_OS_PLATFORM_APPLE || defined(AZ_PLATFORM_LINUX)
-#include <clocale>
-#include <locale>
-
-class LocaleResetter
-{
-public:
-    LocaleResetter()
-    {
-        m_oldLocale = std::setlocale(LC_NUMERIC, "C");
-    }
-    ~LocaleResetter()
-    {
-        std::setlocale(LC_NUMERIC, m_oldLocale);
-    }
-
-private:
-    char* m_oldLocale;
-};
-#define SCOPED_LOCALE_RESETTER LocaleResetter l
-#else
-// noop on Windows
-#define SCOPED_LOCALE_RESETTER
-#endif
+#define SCOPED_LOCALE_RESETTER AZ::Locale::ScopedSerializationLocale localeResetter;
```

Comment:
Do we need this #define anymore? Can we just use a local `localeResetter` instead?

Reply:
I'll replace them all, it won't have an impact on the actual running.